### ISSUE TITLE
OpenCV Error: Assertion failed

### ISSUE THREAD

**Issue by BearVic** [yun-liu/BING#2]
I have successfully installed opencv3 and extracted the data following the instructions, but got these error(I ran successfully for another opencv project removeRedEyes.cpp):

Load annotations finished
Dataset:'../VOC2007/' with 2501 training and 4952 testing
WinRecall.m Base = 2, W = 8, NSS = 2, perSz = 130
OpenCV Error: Assertion failed (0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows) in Mat, file /build/opencv-zcaJjh/opencv-3.2.0+dfsg/modules/core/src/matrix.cpp, line 522
OpenCV Error: Assertion failed (0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows) in Mat, file /build/opencv-zcaJjh/opencv-3.2.0+dfsg/modules/core/src/matrix.cpp, line 522
OpenCV Error: Assertion failed (0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows) in Mat, file /build/opencv-zcaJjh/opencv-3.2.0+dfsg/modules/core/src/matrix.cpp, line 522
OpenCV Error: Assertion failed (0 <= roi.x && 0 <= roi.width && roi.x + roi.width <= m.cols && 0 <= roi.y && 0 <= roi.height && roi.y + roi.height <= m.rows) in Mat, file /build/opencv-zcaJjh/opencv-3.2.0+dfsg/modules/core/src/matrix.cpp, line 522
terminate called after throwing an instance of 'terminate called after throwing an instance of 'cv::Exception'
terminate called recursively
cv::Exception'
terminate called recursively
Aborted (core dumped)

Could you please tell me what is wrong?
Thanks!
@yun-liu

**Comment by wangning316** [comment]
I also encountered this problem when executing the project. Is there anything not set in my system？ @yun-liu

**Comment by wangning316** [comment]
I did it myself，I found the cause of the error and merged the training set and test set into.. \JPEGImages program runs correctly because... The description in \VOC2007\ImageSets\Main is for all images
